runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -221,6 +235,106 @@ private ClusterStatus getClusterHealth(Layout layout, Set<String> peerResponsive
         return peerResponsiveNodes;
     }
 
+    /**
+     * Attempts to fetch the layout with the highest epoch.
+     * This does not consume the corfu runtime layout as that can get stuck in an infinite retry
+     * loop.
+     *
+     * @param layoutServers Layout Servers to fetch the layout from.
+     * @return Latest layout from the servers or null if none of them responded with a layout.
+     */
+    private Layout getHighestEpochLayout(List<String> layoutServers) {
+        Layout layout = null;
+        Map<String, CompletableFuture<Layout>> layoutFuturesMap = new HashMap<>();
+        // Get layout futures for layout requests to all layout servers.
+        for (String server : layoutServers) {
+            try {
+                // Router creation can throw a NetworkException.
+                IClientRouter router = runtime.getRouter(server);
+                layoutFuturesMap.put(server,
+                        new LayoutClient(router, Layout.INVALID_EPOCH).getLayout());
+            } catch (NetworkException e) {
+                log.error("getClusterStatus: Exception encountered connecting to {}. ",
+                        server, e);
+                CompletableFuture<Layout> cf = new CompletableFuture<>();
+                cf.completeExceptionally(e);
+                layoutFuturesMap.put(server, cf);
+            }
+        }
+
+        // Wait on the Completable futures and retain the layout with the highest epoch.
+        for (Entry<String, CompletableFuture<Layout>> entry : layoutFuturesMap.entrySet()) {
+            try {
+                Layout nodeLayout = entry.getValue().get();
+                log.debug("Server:{} responded with: {}", entry.getKey(), nodeLayout);
+
+                if (layout == null || nodeLayout.getEpoch() > layout.getEpoch()) {
+                    layout = nodeLayout;
+                }
+            } catch (InterruptedException ie) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)